roles/manage_dbserver/tasks/manage_postgres_params.yml

```diff
@@ -37,9 +37,46 @@
     - pg_postgres_conf_params|length > 0
     - params.changed
     - not params_restart_required
+    - not use_patroni
+
+- name: Reload patroni postgres
+  ansible.builtin.shell: >
+    {{ patroni_bin_dir }}/patronictl -c {{ patroni_config_file }} \
+        reload {{ pg_instance_name }} {{ inventory_hostname }} --force
+  args:
+    executable: /bin/bash
+  register: patronictl_exec
+  changed_when: patronictl_exec.rc == 0
+  failed_when: patronictl_exec.rc != 0
+  when:
+    - pg_postgres_conf_params|length > 0
+    - params.changed
+    - not params_restart_required
+    - use_patroni
+  become_user: "{{ pg_owner }}"
+  become: true
+
+- name: Restart patroni postgres
+  ansible.builtin.shell: >
+    {{ patroni_bin_dir }}/patronictl -c {{ patroni_config_file }} \
+        restart {{ pg_instance_name }} {{ inventory_hostname }} --force
+  args:
+    executable: /bin/bash
+  register: patronictl_exec
+  changed_when: patronictl_exec.rc == 0
+  failed_when: patronictl_exec.rc != 0
+  when:
+    - pg_postgres_conf_params|length > 0
+    - params.changed
+    - params_restart_required
+    - use_patroni
+  become_user: "{{ pg_owner }}"
+  become: true
 
 - name: Gather services facts
   ansible.builtin.service_facts:
+  when:
+    - not use_patroni
```

Comment:
I think we must gather services facts in all cases. If we don't do that, the following steps including `when` conditions based on `ansible_facts.services` will raise an error when evaluating the conditions.

Reply:
Fixed